### DISCUSSION TITLE
Nice Html Report with Xunit

### DISCUSSION THREAD

**Discussion by mufasalg0** [xunit/xunit#2596] [General]
Hello I have a xUnit testing framework for end to end testing.. One thing I can not figure out is generating nice HTML test results like in this example:

https://www.google.com/search?q=selenium+nice+html+test+results&rlz=1C1RXQR_en&source=lnms&tbm=isch&sa=X&ved=2ahUKEwjT6df8zev6AhVhPH0KHdiyCD4Q_AUoAXoECAEQAw&biw=1920&bih=937&dpr=1#imgrc=JcKrVxP-hu8ahM

All the plugins I have tried failed, I am really struggling with this issue. Could someone please help me?

Thank you!

**Comment by bartelink** [comment]
Perhaps try stack overflow too?

**Comment by mufasalg0** [comment]
I did but no answer.

**Comment by bartelink** [comment]
As noted [over here](https://github.com/xunit/xunit/issues/1200#issuecomment-1286631376), while I appreciate you're frustrated and just trying to get things solved, cross posting doesn't help anyone.

I'd suggest providing links to the stack overflow questions here too, and expanding on 
> have a xUnit testing framework for end to end testing.

(I get you want reports but you can't expect people to read your mind as to what your overall needs are and what "have a xUnit testing framework" means)

**Comment by mufasalg0** [comment]
I appreciate the feedback you have given.
Here is my post.
https://stackoverflow.com/questions/74129135/generating-html-testresults-with-xunit?noredirect=1#comment130889088_74129135

**Reply by bartelink** [reply]
IIRC the Xunit HTML report is just a stylesheet over the XML output. In general it's optimised for surfacing failures more than the stuff that you should have lots of - passing tests that run in an instant. The top hit on a basic search gives an example in the direction of what you're asking.
https://www.google.com/search?q=xunit+html+report

Not sure your question on SO is answerable as it is 

I reckon the choices are either: 
- find a rendering that you like
- use the Xunit standard one
- fork the Xunit standard one

(I personally have zero interest in test reports that tell me about the accomplishments of passing tests - the only thing I care about is narrowing down failures with clear stack traces; if the CI system has stats that help me identify whether a test is flaky, that's also a big win)

I'll step away now as I have no actual (interest or) expertise in this space....

**Comment by bradwilson** [comment]
As suggested elsewhere, there are essentially two answers today:

1. If you're using our runners (meaning, something like `xunit.console.exe`), then you can either use the `-html` switch to get our HTML report, or the `-xml` switch to get our XML report which you could then later transform either with your own XSL-T or with some third party tool.
2. If you're using the VSTest runner (meaning, `dotnet test` or `vstest.console.exe`), then you can generate a TRX output report (which is their own XML format), and then transform that with your own XSL-T or with some third party tool.

The instructions for option 2 should be available from the VSTest team.
